lib/riemann.js

```diff
@@ -9,8 +9,36 @@ var riemann = require("riemann");
  * Optional:
  * transport, flush, packet
  */
+
+
 function Riemann(startupTime, config, emitter) {
   this.config = config.riemann;
+  this.emitter = emitter;
+
+  if (this.config.transport && this.config.transport === "tcp") {
+    setInterval(this.reconnectIfNeeded.bind(this), this.config.reconnectInterval);
```

Comment:
It would be a good idea to ensure that `reconnectInterval` is defined as well. That would allow the end user to only enable this functionality if desired.